enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,995 @@
+  // If nodeNetwork is not provided, default selection
+  // will select the first address that is not a Localhost address.
+  // +kubebuilder:validation:Optional
+  NodeNetworking VSpherePlatformNodeNetworking `json:"nodeNetworking,omitempty"`
+}
+```
+
+### vSphere Machine - VSphereMachineProviderSpec change
+
+Including the change to the vSphere Infrastructure and install platform
+spec we need to include a new parameter to the 
+`VSphereMachineProviderSpec` - `FailureDomain`. This
+string will associate the Machine to the FailureDomain that is defined
+via the install-config platform spec.
+
+```golang
+type VSphereMachineProviderSpec struct {
+
+  // ...
+
+	// CloneMode specifies the type of clone operation.
+	// The LinkedClone mode is only support for templates that have at least
+	// one snapshot. If the template has no snapshots, then CloneMode defaults
+	// to FullClone.
+	// When LinkedClone mode is enabled the DiskGiB field is ignored as it is
+	// not possible to expand disks of linked clones.
+	// Defaults to LinkedClone, but fails gracefully to FullClone if the source
+	// of the clone operation has no snapshots.
+	// +optional
+	CloneMode CloneMode `json:"cloneMode,omitempty"`
+
+
+	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Infrastructure spec.
+	// For this infrastructure provider, the name is equivalent to the name of the VSphereDeploymentZone.
+	FailureDomain *string `json:"failureDomain,omitempty"`
+}
+```
+
+### CCCMO
+
+The CCCMO will need to be modified to support the VSpherePlatformSpec.
+It will need to be able to use the existing INI-based cloud-config spec
+and also translate that configuration to YAML-based cloud-config.
+
+### installer
+
+#### Platform Spec
+
+The platform spec needs to be modified to support our initial goals of multiple
+datacenters (regions) and clusters (zones) and vcenters. This platform spec
+design is based on the changes suggested in openshift/api, Cluster API for vSphere
+and out-of-tree CCM.
+
+We are adding multiple additional parameters to the Platform struct:
+
+- `VCenters`
+- `DeploymentZones`
+- `FailureDomains`
+
+`VCenters` will contain the connections and configuration for each vCenter
+that is required for the out-of-tree CCM. Note: Only a single vCenter will
+be supported by this effort. While the out-of-tree CCM supports multiple vCenters
+the out-of-tree CSI does not.
+
+`FailureDomains` will define the configuration of a region, zone, topology and
+the topology type. `Topology` defines the vCenter objects that make up a region or zone
+including `Datacenter`, `ComputeCluster`, `Hosts`, `Networks` and a `Datastore`.
+
+`DeploymentZones` will define the association between a FailureDomain, vcenter and
+PlacementConstraints - which include Resource Pools and Folders.
+
+The existing platform spec vcenter parameters will be deprecated
+but _not_ removed or remove support for using those parameters.
```

Comment:
I think we need to expand on this: 
- what does it mean to be deprecated but not remove or remove support?
- what is the relationship between `vcenter` and `vcenters`? are they mutually exclusive? if so, we need to add validation in openshift/installer#5911